core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,307 @@
+    try {
+      StringWriter writer = new StringWriter();
+      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
+      if (pretty) {
+        generator.useDefaultPrettyPrinter();
+      }
+      toJson(type, defaultValue, generator);
+      generator.flush();
+      return writer.toString();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static void toJson(Type type, Object javaDefaultValue, JsonGenerator generator)
+      throws IOException {
+    switch (type.typeId()) {
+      case BOOLEAN:
+        generator.writeBoolean((boolean) javaDefaultValue);
+        break;
+      case INTEGER:
+        generator.writeNumber((int) javaDefaultValue);
+        break;
+      case LONG:
+        generator.writeNumber((long) javaDefaultValue);
+        break;
+      case FLOAT:
+        generator.writeNumber((float) javaDefaultValue);
+        break;
+      case DOUBLE:
+        generator.writeNumber((double) javaDefaultValue);
+        break;
+      case DATE:
+        generator.writeString(DateTimeUtil.formatEpochDays((int) javaDefaultValue));
+        break;
+      case TIME:
+        generator.writeString(DateTimeUtil.formatTimeOfDayMicros((long) javaDefaultValue));
+        break;
+      case TIMESTAMP:
+        generator.writeString(DateTimeUtil.formatEpochTimeMicros(
+            (long) javaDefaultValue,
+            ((Types.TimestampType) type).shouldAdjustToUTC()));
+        break;
+      case STRING:
+        generator.writeString((String) javaDefaultValue);
+        break;
+      case UUID:
+        generator.writeString(javaDefaultValue.toString());
+        break;
+      case FIXED:
+      case BINARY:
+        generator.writeString(BaseEncoding.base16().encode(ByteBuffers.toByteArray(((ByteBuffer) javaDefaultValue))));
+        break;
+      case DECIMAL:
+        generator.writeNumber((BigDecimal) javaDefaultValue);
+        break;
+      case LIST:
+        List<Object> defaultList = (List<Object>) javaDefaultValue;
+        Type elementType = type.asListType().elementType();
+        generator.writeStartArray();
+        for (Object elementDefault : defaultList) {
+          toJson(elementType, elementDefault, generator);
+        }
+        generator.writeEndArray();
+        break;
+      case MAP:
+        Map<Object, Object> defaultMap = (Map<Object, Object>) javaDefaultValue;
+        List<Object> keyList = Lists.newArrayListWithExpectedSize(defaultMap.size());
+        List<Object> valueList = Lists.newArrayListWithExpectedSize(defaultMap.size());
+        Type keyType = type.asMapType().keyType();
+        Type valueType = type.asMapType().valueType();
+
+        for (Map.Entry<Object, Object> entry : defaultMap.entrySet()) {
+          keyList.add(entry.getKey());
+          valueList.add(entry.getValue());
+        }
+        generator.writeStartObject();
+        generator.writeFieldName("keys");
```

Comment:
Strings that are part of the serialization format should be constants at the top, like the other parsers use.